index.js

```diff
@@ -154,6 +165,37 @@ function inject (bot) {
 
   bot.on('physicTick', monitorMovement)
 
+  /**
+   * Finds blocks intersected by entity bounding boxes
+   * and sets the number of ents intersecting in a dict
+   * @param {object} entIntersections Dict to write to. Follows d[y][x][z] = #ents
+   */
+  function updateCollisionIndex (entIntersections) {
+    for (const ent of Object.values(bot.entities)) {
+      // TODO: Maybe add a seperate modifier for hostiles or an 'entitiesToAvoid' list?
+      if (ent !== bot.entity && passableEnts.includes(ent.name)) {
+        const entSquareRadius = ent.width / 2.0
+        const minY = Math.floor(ent.position.y)
+        const maxY = Math.ceil(ent.position.y + ent.height)
+        const minX = Math.floor(ent.position.x - entSquareRadius)
+        const maxX = Math.ceil(ent.position.x + entSquareRadius)
+        const minZ = Math.floor(ent.position.z - entSquareRadius)
+        const maxZ = Math.ceil(ent.position.z + entSquareRadius)
+
+        for (let y = minY; y < maxY; y++) {
+          entIntersections[y] = entIntersections[y] ?? {}
+          for (let x = minX; x < maxX; x++) {
+            entIntersections[y][x] = entIntersections[y][x] ?? {}
+            for (let z = minZ; z < maxZ; z++) {
+              entIntersections[y][x][z] = entIntersections[y][x][z] ?? 0
+              entIntersections[y][x][z]++ // More ents = more weight
+            }
+          }
+        }
```

Comment:
Have you tried it with just indexing the object like so ```entIntersections[`${x}${y}${z}`]````this might actually be faster then having to index nested objects.

Reply:
Agreed, that is a lot neater. Some surface level tests also say that it is ~10% faster